theano/gpuarray/c_code/ctc_wrapper.c

```diff
@@ -0,0 +1,293 @@
+#section support_code
+
+typedef struct ctc_context {
+    struct ctcOptions options;
+    PyGpuArrayObject * workspace;
+    int * input_lengths;
+    int * flat_labels;
+    int * label_lengths;
+} ctc_context_t;
+
+void ctc_context_init(ctc_context_t * context, PyGpuContextObject * gpu_context)
+{
+    memset(&(context->options), 0, sizeof(struct ctcOptions));
+    context->options.loc = CTC_GPU;
+
+    // Get CUDA function pointer to obtain stream
+    CUstream (*getstream_func_ptr)(void *) = (CUstream (*)(void *)) gpuarray_get_extension( "cuda_get_stream" );
+    context->options.stream = getstream_func_ptr(gpu_context->ctx);
```

Comment:
You should use the NULL stream here or add proper synchronization for all your GPU inputs/outputs in the calling code.